frontend/packages/dev-console/src/components/pipelines/pipeline-overview/TriggerLastRunButton.tsx

```diff
@@ -0,0 +1,27 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { connect } from 'react-redux';
+import { impersonateStateToProps } from '@console/internal/reducers/ui';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import { useAccessReview } from '@console/internal/components/utils';
+import { Button } from 'patternfly-react';
+import { rerunPipeline } from '../../../utils/pipeline-actions';
+import { PipelineModel } from '../../../models';
+
+type TriggerLastRunButtonProps = {
+  pipeline: K8sResourceKind;
+  impersonate?;
```

Comment:
I know it's not really typed anywhere - but we should definitely look to avoid untyped props. Could you check to see if you can type this?